### PR TITLE
Fix delete goods bugs

### LINKED ISSUE #110: Deletion of goods should be supplier specific
Current deletion logic allows deletion via goods name. Hence, if `Bread` has been supplied by two suppliers, `deletegoods gn/Bread` will result in both suppliers records being deleted. As such, delete goods should consider both goods name and supplier name to ensure data integrity.

### PR DESCRIPTION
**Pull request by yanhwee**
Fixes #110 and #101.

### CODE REVIEW

**Comment by jialegitgud** [comment]
LGTM, but yes, would be good to include higher code coverage for the relevant classes.

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -6,7 +6,8 @@
 public class CliSyntax {
 
     /* Prefix definitions for Persons */
-    public static final Prefix PREFIX_NAME = new Prefix("n/");
+    public static final Prefix
```
Comment (stephaneleong) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:9]: preferred to keep variable declaration in one line

File: src/main/java/seedu/address/logic/commands/DeleteGoodsCommand.java [hunk 2]
```diff
@@ -2,35 +2,53 @@
...
 
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.goods.GoodsName;
+import seedu.address.model.goodsreceipt.GoodsReceipt;
+import seedu.address.model.person.Name;
 
 /**
  * Deletes all goods that match the given goodsName.
```
Comment (stephaneleong) [inline src/main/java/seedu/address/logic/commands/DeleteGoodsCommand.java:15]: DeleteGoodsCommand javadocs needs to be updated. should include suppliername if command has changed

File: src/main/java/seedu/address/model/ReceiptLog.java [hunk 3]
```diff
@@ -85,6 +85,10 @@ public void removeIf(Predicate<? super GoodsReceipt> predicate) {
         receipts.removeIf(predicate);
     }
 
+    public void remove(GoodsReceipt receipt) {
```
Comment (stephaneleong) [inline src/main/java/seedu/address/model/ReceiptLog.java:88]: doesn't deleteReceipt exist already at line 64?

Comment (stephaneleong) [reply src/main/java/seedu/address/model/ReceiptLog.java:88]: can relook if this method is necessary

File: src/main/java/seedu/address/model/Model.java [hunk 4]
```diff
@@ -119,10 +118,14 @@ public interface Model {
     void addGoods(GoodsReceipt goodsReceipt);
 
     /**
-     * Deletes all the goods which match the GoodsName given.
-     * @param goodsName The GoodsName indicating the goods to be deleted.
+     * Deletes the given good receipt.
      */
-    void deleteGoods(GoodsName goodsName);
+    void deleteGoodsReceipt(GoodsReceipt goodsReceipt);
```
Comment (stephaneleong) [inline src/main/java/seedu/address/model/Model.java:123]: maybe consider keeping naming convention consistent. it is confusing to have addGoods and deleteGoodsReceipt be part of the same CRUD model yet have different naming

**Review by stephaneleong — COMMENTED** [review]
overal good changes to logic but still need to consider some changes before merging

File: src/main/java/seedu/address/model/ReceiptLog.java (hunk 3 above)
Comment (yanhwee) [reply src/main/java/seedu/address/model/ReceiptLog.java:88]: You are right!

File: src/main/java/seedu/address/model/Model.java (hunk 4 above)
Comment (yanhwee) [reply src/main/java/seedu/address/model/Model.java]: Alright I have changed `deleteGoodsReceipt` to `deleteGoods` (it still takes in the `GoodsReceipt`)

File: src/main/java/seedu/address/logic/commands/DeleteGoodsCommand.java (hunk 2 above)
Comment (yanhwee) [reply src/main/java/seedu/address/logic/commands/DeleteGoodsCommand.java]: 👍

**Pull request merged**